File: LearningObjectOrientedExercisesCollection/LP02/Student.java
package com.LearningObjectOrientedExercisesCollection.LP02;
/**
 * static 修饰属性
 * */
public class Student {
    //属性
    String name;
    int age;
    static String school;

    public static void main(String[] args) {
        //可以用于定义对象中固定值
        Student.school="工业经济";
        //创建学生对象
        Student s1 = new Student();
        s1.name="zhangsan";
        s1.age=23;
        System.out.println("姓名"+s1.name+"年龄"+s1.age+"学校"+school);

        Student s2=new Student();
        s2.name="lili";
        s2.age=20;
        System.out.println("姓名"+s2.name+"年龄"+s2.age+"学校"+school);


    }
}

File: com/Chapter01/demo01/Student.scala
package com.Chapter01.demo01

/**
 * 当一个类和另一个对象同名且定义在同一个文件中，
 * 这个类称为半生类而这个对象称为伴生对象。两者属于伴生关系。可以互相访问对象的私有成员
 */

class Student(xname:String,xage :Int) {
  private val name=xname;
  val age =xage

  def sayName()={
    //类中获取Object中私有变量
    s"Object name is ${Student.yname}"
  }
}

object Student {
  val yname="ZhangSan";
  def main(args: Array[String]): Unit = {
    val student = new Student("LiSi", 23)
   //获取类中有变量
    println(student.name)
    println(student.sayName())


  }
}





